bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/tools/WaitForTopicValue.java

```diff
@@ -68,7 +68,7 @@ public void stop() {
      */
     public @Nullable String waitForTopicValue(int timeoutInMS) {
         try {
-            return subscripeFuture.thenCompose(b -> future).get(timeoutInMS, TimeUnit.MILLISECONDS);
+            return subscribeFuture.thenCompose(b -> future).get(timeoutInMS, TimeUnit.MILLISECONDS);
```

Comment:
seems a bit wasteful to create a new compose future every time. Could you cache `subscribeFuture.thenCompose(b -> future)` in a field?

Reply:
Ok, I will modify it, but keep in mind that the WaitForTopicValue is a one-time use class and a new object should be created in each use. What would be the benefit?